content/posts/2021/reana-client-python-api.md

```diff
@@ -0,0 +1,330 @@
+
+```python
+from reana_client.api.client import upload_to_server
+
+abs_path_to_input_files = [os.path.abspath(f) for f in my_inputs['files']]
+upload_to_server(my_workflow_name, abs_path_to_input_files, my_reana_token)
+```
+
+### Starting workflow
+
+At this point we have done all what is needed to start our workflow in REANA.
+To do it with REANA client we could call [start](https://reana-client.readthedocs.io/en/maint-0.7/#reana-client-start)
+command. But in this case it can be started by simply calling `start_workflow`
+function and defining workflow name and access token. Additionally, we
+can also override workflow parameters while starting the workflow, but let's
+keep the original ones for now:
+
+```python
+from reana_client.api.client import start_workflow
+
+start_workflow(my_workflow_name, my_reana_token, {})
+```
+
+Output:
+
+```python
+{
+  'message': 'Workflow submitted.',
+  'run_number': 1,
+  'status': 'queued',
+  'user': '1d27737b-5b80-4da2-87bb-816a20e110bd',
+  'workflow_id': '6cd613eb-f2fb-411b-9601-c89599925759',
+  'workflow_name': 'root6-roofit'
+}
+```
+
+### Checking workflow status
+
+After starting our workflow we might want to check its current status.
+To inspect the workflow status using REANA client we could use [status](https://reana-client.readthedocs.io/en/maint-0.7/#reana-client-status)
+command. Similarly, it can be achieved with `get_workflow_status` function.
+Since it will take couple of seconds for all the computations to be finished
+while executing our workflow, we will use a loop to wait and regurlary check
+what is the current status of our workflow:
+
+```python
+import time
+from reana_client.api.client import get_workflow_status
+
+while True:
+    status_details = get_workflow_status(my_workflow_name, my_reana_token)
+    print('Current status: ', status_details['status'])
+    if status_details['status'] == 'finished':
+        break
+    time.sleep(1)
+```
+
+Output:
+
+```python
+Current status:  running
+...
+Current status:  finished
+```
+
+### Checking workflow logs
+
+If our workflow has failed or in case we want to debug or investigate how our
+workflow works, we may want to display the logs of our workflow. To do that
+using REANA client we could use [logs](https://reana-client.readthedocs.io/en/maint-0.7/#reana-client-logs)
+command. In our case the logs can be fetched with `get_workflow_logs` function
+where we only need to specify workflow name and access token. Optionally, we
+can add such parameters as a list of `steps` whose logs we are interested in
+or `page` and `size` parameters to paginate the logs output if it gets too
+verbose. Workflow logs can be retrieved as follows:
+
+```python
+import json
+from reana_client.api.client import get_workflow_logs
+
+logs = get_workflow_logs(my_workflow_name, my_reana_token)
+logs_json = json.loads(logs['logs'])
+print(json.dumps(logs_json, indent=2))
+```
+
+Output:
+
+```python
+{
+  'workflow_logs': ...,
```

Comment:
An example comment: The three dots here make the example a bit opaque. What about doing, as an example, something smaller, notably how to get logs for the second "fitdata" step, and print first say 15 lines of the log?  A little Python loop over steps, localising "gendata" name, and reading logs and spliting by newline and printing first say 15 lines of the "fitdata" step?  Or, even better, print out the last 15 lines of the step, as if seeking for the end of progress... This would make a nice transition to making the plot, because readers will see the plot being generated in the last 15 lines of the fitdata step. I think this would be a rather practical example, where we could even show the exact output, and it would also look better than the json dump in the ipython notebook itself?